pkg/network/node/networkpolicy.go

```diff
@@ -371,6 +395,31 @@ func (np *networkPolicyPlugin) updateMatchCache(npns *npNamespace) {
 	}
 }
 
+func (np *networkPolicyPlugin) updatePodMatchCache(pod, oldPod *corev1.Pod) {
+	var potentialMatches map[string]*npPodCacheEntry
+	if pod != nil {
+		potentialMatches = np.podMatchCache[pod.Namespace]
+	}
+
+	//pod deletion case
+	if pod == nil && oldPod != nil {
+		potentialMatches = np.podMatchCache[oldPod.Namespace]
```

Comment:
You could simplify this a little by just passing the namespace name in from the caller so that then you don't have to worry about which of `pod` or `oldPod` is `nil` here

Reply:
done